### ISSUE TITLE
Observabilty Operator v0.0.13 pod crashing on OpenShift Sandbox Production environment 

### ISSUE THREAD

**Issue by redhatHameed** [rhobs/observability-operator#203]
**Background** 

We are using the Observability Operator with [RHODA Operator](https://github.com/RHEcosystemAppEng/dbaas-operator), for scraping the metrics and remote write to observatorium and installing and configuring it with the code within [dbaas operator](https://github.com/RHEcosystemAppEng/dbaas-operator/blob/main/controllers/reconcilers/providersinstallation/observability-operator.go#L36) and currently deployed with RHODA 0.3.0 release. 

**Issue :** **Observabilty Operator pod crashing on Sandbox (cluster version 4.11.5)**

we have deployed the observabilty operator version 0.0.13 on the sandbox production environment with [Red Hat OpenShift Database Access release 0.3.0](https://github.com/RHEcosystemAppEng/dbaas-operator/tree/release-0.3.0)  and the pod `observability-operator-prometheus-operator`
 start crashing with high memory leakage, as seen in the below screenshot.

![image (6)](https://user-images.githubusercontent.com/41960152/196225395-d3b744ca-a7c3-48dd-9337-3bd7f6dbbe50.png)

![image (7)](https://user-images.githubusercontent.com/41960152/196225485-57f25717-577b-4ad4-89b4-da5e7a38633b.png)

Logs from the pod :

``` 
trace.go:205] Trace[642937274]: "DeltaFIFO Pop Process" ID:alifcsnajmi-dev/builder,Depth:3037,Reason:slow event handlers blocking the queue (17-Oct-2022 14:19:11.527) (total time: 199ms)
Trace[642937274]: [199.502387ms] [199.502387ms] END 
```
Complete trace : 
```
2022-10-17T14:19:01Z	INFO	controller.monitoringstack	Starting Controller	{"reconciler group": "monitoring.rhobs", "reconciler kind": "MonitoringStack"}
2022-10-17T14:19:01Z	INFO	controller.thanosquerier	Starting EventSource	{"reconciler group": "monitoring.rhobs", "reconciler kind": "ThanosQuerier", "source": "kind source: *v1.Deployment"}
2022-10-17T14:19:01Z	INFO	controller.thanosquerier	Starting EventSource	{"reconciler group": "monitoring.rhobs", "reconciler kind": "ThanosQuerier", "source": "kind source: *v1.ServiceAccount"}
2022-10-17T14:19:01Z	INFO	controller.thanosquerier	Starting EventSource	{"reconciler group": "monitoring.rhobs", "reconciler kind": "ThanosQuerier", "source": "kind source: *v1.Service"}
2022-10-17T14:19:01Z	INFO	controller.thanosquerier	Starting EventSource	{"reconciler group": "monitoring.rhobs", "reconciler kind": "ThanosQuerier", "source": "kind source: *v1alpha1.MonitoringStack"}
2022-10-17T14:19:01Z	INFO	controller.thanosquerier	Starting Controller	{"reconciler group": "monitoring.rhobs", "reconciler kind": "ThanosQuerier"}
2022-10-17T14:19:01Z	INFO	thanos-querier	watched MonitoringStack changed, checking for matching querier	{"Monitoring Stack": "redhat-dbaas-operator/dbaas-operator-mso"}
I1017 14:19:09.126333       1 trace.go:205] Trace[419353989]: "DeltaFIFO Pop Process" ID:marikan-dev/workspace9723c3d02de146a8-service,Depth:1227,Reason:slow event handlers blocking the queue (17-Oct-2022 14:19:08.927) (total time: 199ms):
Trace[419353989]: [199.160873ms] [199.160873ms] END
I1017 14:19:09.426261       1 trace.go:205] Trace[777450220]: "DeltaFIFO Pop Process" ID:anuwat-dev/parksmap,Depth:1140,Reason:slow event handlers blocking the queue (17-Oct-2022 14:19:09.126) (total time: 299ms):
Trace[777450220]: [299.397412ms] [299.397412ms] END
I1017 14:19:10.226243       1 trace.go:205] Trace[62086580]: "DeltaFIFO Pop Process" ID:igoorsp-dev/deployer,Depth:4533,Reason:slow event handlers blocking the queue (17-Oct-2022 14:19:10.027) (total time: 100ms):
Trace[62086580]: [100.080392ms] [100.080392ms] END
I1017 14:19:10.427488       1 trace.go:205] Trace[1158964279]: "DeltaFIFO Pop Process" ID:wepzen-dev/deployer,Depth:3955,Reason:slow event handlers blocking the queue (17-Oct-2022 14:19:10.327) (total time: 100ms):
Trace[1158964279]: [100.309422ms] [100.309422ms] END
I1017 14:19:10.927244       1 trace.go:205] Trace[1710066391]: "DeltaFIFO Pop Process" ID:hazemamer-dev/builder,Depth:3669,Reason:slow event handlers blocking the queue (17-Oct-2022 14:19:10.627) (total time: 300ms):
Trace[1710066391]: [300.065372ms] [300.065372ms] END
I1017 14:19:11.726679       1 trace.go:205] Trace[642937274]: "DeltaFIFO Pop Process" ID:alifcsnajmi-dev/builder,Depth:3037,Reason:slow event handlers blocking the queue (17-Oct-2022 14:19:11.527) (total time: 199ms):
Trace[642937274]: [199.502387ms] [199.502387ms] END 
```
OOMKilled status also showed up on the pod between restarts.

we also seen the similar logs on the clusters(non sandbox environment)  but the pod is not crashing as in below screenshot.
![image (9)](https://user-images.githubusercontent.com/41960152/196229821-ed87faec-c622-44bd-9df0-d8b52cbf3841.png)

**Comment by simonpasquier** [comment]
> the pod observability-operator-prometheus-operator start crashing with high memory leakage

I'm not sure that it's a "real" memory leak but most probably, the container gets killed because it hits the memory limits. Can you check the resource limits of the pod? It might also be worth trying with v0.0.14.

**Comment by tremes** [comment]
Yes it seems the limit (https://github.com/rhobs/observability-operator/blob/main/deploy/operator/observability-operator-deployment.yaml#L44) was reached. Should we increase it? I guess the v0.0.14 will be the same.

**Comment by redhatHameed** [comment]
Further clarification issue is cause by deployment/pd of `prometheus-operator` not the observability operator pod, we also tried to reproduce the issue on our cluster by creating 500 namespaces (with 2 user each), and prometheus pod start crashing with oomkilled error.

we tried the latest release of observability operator and same issue faced by `obo-prometheus-operator` pod, and after setting the resource limit         ```- resources:
            limits:
              cpu: 100m
              memory: 1500Mi``` it get fixed on our cluster, but we can't change this into sandbox as it get reverted by obseravabilty operator. 

so it looks the prometheus operator watching objects from all nanmesapce which use more memory and in sandbox environment we don't know actual number of namespaces and what resource required for permanent fix. 

screenshot memory usage from our local cluster of obo-prometheus-operator.
<img width="1061" alt="Screen Shot 2022-10-18 at 1 22 47 PM" src="https://user-images.githubusercontent.com/41960152/196502605-1e67d887-721b-43e4-a0c2-e7d3214d5798.png">

**Comment by simonpasquier** [comment]
I've filed https://github.com/prometheus-operator/prometheus-operator/issues/5410 to track it upstream.

**Comment by sthaha** [comment]
@simonpasquier @redhatHameed how that we have merged 0.64, which makes use of metadata informers, can we close this?

**Comment by simonpasquier** [comment]
sure
